javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql

```diff
@@ -43,6 +43,10 @@ where
     or
     // target is a HTTP URL to a domain on any TLD
     target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+([a-z]+)(:[0-9]+)?/?")
+    or
+    // target is a HTTP URL to a domain on any TLD with path elements, and the check is an includes check
+    check instanceof StringOps::Includes and
+    target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+([a-z]+)(:[0-9]+)?/([a-z0-9-]+/?)*")
```

Comment:
Optional changes (3), pick the ones you want:

- consistency for TLDs
- more permissive paths
- flattening of quantifiers
```suggestion
    target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+" + RegExpPatterns::commonTLD() + "(:[0-9]+)?/[a-z0-9/_-]+")
```

Reply:
I'm going with 2+3.